presto-hive/src/main/java/io/prestosql/plugin/hive/HiveConnectorFactory.java

```diff
@@ -51,6 +56,21 @@ public ConnectorHandleResolver getHandleResolver()
     @Override
     public Connector create(String catalogName, Map<String, String> config, ConnectorContext context)
     {
-        return createConnector(catalogName, config, context, Optional.empty());
+        ClassLoader classLoader = context.duplicatePluginClassLoader();
+        try {
+            Class<?> hiveConnectorFactoryClass = classLoader.loadClass(HIVE_CONNECTOR_FACTORY);
+            checkState(hiveConnectorFactoryClass != null, "Could not load %s from duplicate plugin class loader", HIVE_CONNECTOR_FACTORY);
+
+            Method method = hiveConnectorFactoryClass.getMethod("createConnector", String.class, Map.class, ConnectorContext.class, Optional.class);
+            return (Connector) method.invoke(null, catalogName, config, context, Optional.empty());
```

Comment:
Would be great to exercise this code also outside of product tests.

Reply:
TestHiveConnectorFactory tests this class